### PR TITLE
Fix test failures reported twice

### PR DESCRIPTION
**Pull request by vegaro**
Example https://app.circleci.com/pipelines/gh/RevenueCat/purchases-android/21113/workflows/b4668dab-588c-43ca-a583-c57620f6449a/jobs/181575/steps

Captura de pantalla 2026-04-17 a las 16 35 23

It was finding xml of the tests twice:

```
 * /home/circleci/project/build/test-results-collected/ui/revenuecatui/build/test-results/testDefaultsBc8DebugUnitTest/TEST-com.revenuecat.purchases.ui.revenuecatui.data.PaywallViewModelTest.xml
 * /home/circleci/project/build/test-results-collected/build/test-results-collected/ui/revenuecatui/build/test-results/testDefaultsBc8DebugUnitTest/TEST-com.revenuecat.purchases.ui.revenuecatui.data.PaywallViewModelTest.xml 
  ```
  
The second path has `build/test-results-collected/build/test-results-collected/...`, the copy of the copy. The test fails once, reported twice

---

> [!NOTE]
> **Low Risk**
> Low risk CI-only change that tweaks how JUnit XML artifacts are gathered; main risk is missing test reports if the new `find | xargs cp` pipeline behaves differently in edge cases (e.g., unusual filenames).
> 
> **Overview**
> Prevents CircleCI from re-copying files under `build/test-results-collected` when gathering JUnit XMLs, eliminating duplicate test report entries.
> 
> Replaces the `find ... -exec cp` approach with a pruned `find ... -print0 | xargs -0 cp --parents` pipeline in both the main `test` job and `test-galaxy` job.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit e6a0fdd02cacf1187d5ace8a2efed4609d804284. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: ui/revenuecatui/src/test/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModelTest.kt [hunk 1]
```diff
@@ -716,7 +716,7 @@ class PaywallViewModelTest {
         }
 
         assertThat(state.errorMessage)
-            .isEqualTo("The RevenueCat dashboard does not have a current offering configured.")
+            .isNotEqualTo("The RevenueCat dashboard does not have a current offering configured.")
```
Comment (tonidero) [inline ui/revenuecatui/src/test/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModelTest.kt]: I'm guessing this needs to be reverted? :P

**Review by tonidero — APPROVED** [review]
I think this makes sense. Thanks for fixing!

File: ui/revenuecatui/src/test/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModelTest.kt (hunk 1 above)
Comment (vegaro) [reply ui/revenuecatui/src/test/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModelTest.kt]: yes! I was just testing it worked lol. Wasn't expecting such quick review. Thank you!!!
